core/src/test/scala/org/http4s/parser/AcceptHeaderSpec.scala

```diff
@@ -82,32 +77,30 @@ class AcceptHeaderSpec extends Specification with HeaderParserHelper[Accept] wit
 
     "Parse multiple Types" in {
       // Just do a single type
-      val accept = Accept(`audio/mod`, `audio/mpeg`)
-      parse(accept.value) must be_==(accept)
+      val accept = Accept(MediaRangeAndQValue(`audio/mod`), MediaRangeAndQValue(`audio/mpeg`))
+      parse(accept.value) must be_===(accept)
 
       // Go through all of them
-      val ranges = MediaType.snapshot.values.toArray
-      foreach(0 until (ranges.length-1)) { i =>
-        val subrange = ranges.slice(i, i + 4)
-        val h = Accept(subrange.head, subrange.tail:_*)
-        parse(h.value) must be_==(h)
+      val samples = MediaType.snapshot.values.map(MediaRangeAndQValue(_))
+      foreach(samples.sliding(4).toArray) { sample =>
+        val h = Accept(sample.head, sample.tail.toSeq:_*)
+        parse(h.value) must be_===(h)
       }
     }
 
     "Deal with q and extensions" in {
       val value = "text/*;q=0.3, text/html;q=0.7, text/html;level=1"
-      parse(value) must be_==(Accept(
-        `text/*`.withQValue(q(0.3)),
-        `text/html`.withQValue(q(0.7)),
-        `text/html`.withExtensions(Map("level" -> "1"))
+      parse(value) must be_===(Accept(
+        MediaRangeAndQValue(`text/*`,q(0.3)),
```

Comment:
Is there any reason a `MediaRange` shouldn't have a `.withQValue` that returns a `MediaRangeAndQValue`?  The new structure seems reasonable, but I think I preferred that old syntax.